tensorrt_llm/_torch/pyexecutor/kv_cache_connector.py

```
@@ -0,0 +1,549 @@
+        num_tokens, load_kv_async = self._run_on_leader(
+            lambda: self.scheduler.get_num_new_matched_tokens(
+                request, num_computed_tokens))
+
+        if num_tokens == 0 and load_kv_async:
+            raise RuntimeError(
+                "load_kv_async must be False when num_tokens is 0!")
+
+        # TODO(jthomson04): This part is a bit ugly.
+        # When the connector indicates that a request will be loaded asynchronously, we need to suspend it's execution.
+        # This is problematic, since at the point when this function is called, the request has already been scheduled!
+        # Because of this, we need to remove it from our list of scheduled requests (see `take_scheduled_requests_pending_load`).
```

Ruff (0.12.2)

Line too long (129 > 120)

(E501)